### DISCUSSION TITLE
GPT 4 + Moving / New Bot

### DISCUSSION THREAD

**Discussion by cooperlees** [Kav-K/GPTDiscord#224] [General]
Howdy - I've done no research here - But how do we want to move the bot or make a new bot for GPT4? Which way is preferred and why?

## Info
- https://openai.com/product/gpt-4
- https://openai.com/waitlist/gpt-4-api

Thanks!

**Comment by Kav-K** [comment]
Hi cooper!

I'll be making GPT4 integration available **as soon as it's out of waitlist**. 

The plan is to simply integrate the GPT4 model right into the regular /gpt converse flow (now with image multimodal support) and into the other functionalities like /gpt ask and etc. The API is very similar to the ChatGPT API so this will be very simple

**Reply by cooperlees** [reply]
Awesome, thanks. I thought you'd be on top of it.

So, if we add an API token for GPT4 once it exists in config + supported, will it be preferred? Or will we have to be explicit and remove GPT3 API token?

**Reply by Kav-K** [reply]
The token will be the same OpenAI API token, and the only thing you'd have to do is switch the model to GPT4 using /system settings

**Reply by Kav-K** [reply]
Just like the switch to ChatGPT, everything will be handled behind the scenes

**Reply by acxhub** [reply]
Hi @Kav-K I think ChatGPT is now out of the Waitlist correct me if I'm wrong.

**Reply by Kav-K** [reply]
Hi, ChatGPT and GPT4 have both been supported fully for a while now @acxhub

**Reply by acxhub** [reply]
Hi @Kav-K thanks for the prompt response, how do i switch to the GPT4 model?
